ui/src/extensionVariables.ts

```diff
@@ -49,4 +51,25 @@ export function registerUIExtensionVariables(extVars: UIExtensionVariables): voi
     assert(extVars.ui, 'registerUIExtensionVariables: Missing ui');
 
     ext = Object.assign(extVars, { _internalReporter: createTelemetryReporter(extVars.context) });
+
+    registerErrorHandler(handleEntryNotFound);
+}
+
+/**
+ * Long-standing issue that is pretty common for all Azure calls, but can be fixed with a simple reload of VS Code
+ * https://github.com/microsoft/vscode-azure-account/issues/53
+ */
+async function handleEntryNotFound(context: types.IErrorHandlerContext): Promise<void> {
+    if (parseError(context.error).message === 'Entry not found in cache.') {
```

Comment:
Is the message reliably exactly this, or should a regex or something a little more flexible be used?

Reply:
Unfortunately this is extremely reliable haha. The exact source is [here](https://github.com/AzureAD/azure-activedirectory-library-for-nodejs/blob/dev/lib/token-request.js#L526), there's been an open PR to fix the root cause [here](https://github.com/AzureAD/azure-activedirectory-library-for-nodejs/pull/188) for several years, and per [this comment](https://github.com/AzureAD/azure-activedirectory-library-for-nodejs/issues/199#issuecomment-566266974), I don't think they'll ever update the package again. I think the only way to fix the root cause is to move to the new auth library they mentioned, but that's up to the VS Code peeps in the Azure Account extension